src/eradius_server.erl

```diff
@@ -391,10 +387,45 @@ inc_reply_counter(discnak, NasProp) ->
 inc_reply_counter(_Cmd, _NasProp) ->
     ok.
 
+inc_request_counter_accounting(accreq = Cmd, NasProp, Ms, #radius_request{attrs = Attrs}) ->
+    AccReqs = build_counter_accounting(Cmd, Attrs, [{accounting, accountRequests}]),
+    lists:foreach(fun({Account, AccountRequest}) ->
+        eradius_metrics:update_nas_request(Account, NasProp#nas_prop.metrics_info, Ms),
+        eradius_counter:inc_request_counter(AccountRequest, NasProp)
+    end, AccReqs);
+inc_request_counter_accounting(_, _, _, _) ->
+    ok.
+
+inc_response_counter_accounting(accresp = Cmd, NasProp, #radius_request{attrs = Attrs}) ->
+    AccReqs = build_counter_accounting(Cmd, Attrs, [{accounting, accountResponses}]),
+    lists:foreach(fun({Account, AccountResponses}) ->
+        eradius_metrics:update_nas_response(Account, NasProp#nas_prop.metrics_info),
+        eradius_counter:inc_reply_counter(AccountResponses, NasProp)
+    end, AccReqs);
+inc_response_counter_accounting(_, _, _) ->
+    ok.
+
 %% @TODO: extend for other failures
 inc_discard_counter(malformed, NasProp, Ms) ->
     eradius_metrics:update_nas_request(malformed, NasProp#nas_prop.metrics_info, Ms),
     eradius_counter:inc_counter(malformedRequests, NasProp);
 inc_discard_counter(_Reason, NasProp, Ms) ->
     eradius_metrics:update_nas_request(dropped, NasProp#nas_prop.metrics_info, Ms),
     eradius_counter:inc_counter(packetsDropped, NasProp).
+
+build_counter_accounting(_, [], Acc) ->
+    Acc;
+build_counter_accounting(accreq, [{{_, ?RStatus_Type, _, _, _}, ?RStatus_Type_Start} | _], Acc) ->
```

Comment:
`{_, ?RStatus_Type, _, _, _}` is a #attribute record. So matching with `#attribute{id = ...}` should work.

Also this whole list logic could be rewritten with a ets:match_spec_run/2 on the list of attributes.

Additional bonus points will be given for caching the compile match_spec in persistent_term ;-)

Reply:
If I correct got your point - Done.